### PR TITLE
chore: Remove evalError & values from dataTree

### LINKED ISSUE #14710: [Task]: Remove `__evaluation__.evaluatedValues` from dataTree reducing the size of it
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

We maintain the evaluatedValues of each entity inside dataTree object at propertyPaths and also inside `__evaluation__.evaluatedValues`.

This values are duplicate and after each evaluation when we run 

`diff(oldDataTree,dataTree)` this adds 50% more effort because of duplicate diffs.

- [ ] We make sure to remove usage of `__evaluation__.evaluatedValues` from JSONForm
- [ ] Remove the `__evaluation__.evaluatedValues`
- [ ] measure the performance improvement

Internal discussion on this - https://theappsmith.slack.com/archives/C02K0SZQ7V3/p1655741672536679

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

Appsmith internally maintains the data of all the entities as a collection called dataTree. This data tree stores evaluated values and errors ( evaluated props ) in entity object with key `__evaluation__`. 

In this PR, we move the evaluated props out of the dataTree for worker code to make `deepCloning` and dataTree sanitization i.e., `JSON.parse(JSON.stringify(dataTree))` faster. 

TL;DR

Trimming dataTree object size by evaluatedProps to make evaluation faster.

Fixes #14710 

## Type of change

Performance change

- Manual
- [ ] https://github.com/appsmithorg/TestSmith/issues/2124

- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [x] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/workers/Evaluation/dataTreeUtils.ts [hunk 1]
```diff
@@ -39,16 +41,34 @@ export function createUnEvalTreeForEval(unevalTree: UnEvalTree) {
  */
 export function makeEntityConfigsAsObjProperties(
   dataTree: DataTree,
-  option = {} as { sanitizeDataTree: boolean },
-) {
-  const { sanitizeDataTree = true } = option;
+  option = {} as {
```
Comment (ohansFavour) [inline app/client/src/workers/Evaluation/dataTreeUtils.ts:44]: Do we expect the options list to grow?, if so, create an interface and use Partial<>

File: app/client/src/workers/Evaluation/dataTreeUtils.ts [hunk 2]
```diff
@@ -39,16 +41,34 @@ export function createUnEvalTreeForEval(unevalTree: UnEvalTree) {
...
     newDataTree[entityName] = { ...entityConfig, ...entity };
   }
-  return sanitizeDataTree
+  const dataTreeToReturn = sanitizeDataTree
     ? JSON.parse(JSON.stringify(newDataTree))
     : newDataTree;
+
+  if (!evalValuesAndErrors) return dataTreeToReturn;
+
+  for (const [entityName, entityEvalValues] of Object.entries(
```
Comment (ohansFavour) [inline app/client/src/workers/Evaluation/dataTreeUtils.ts:62]: Need clarification:

Doesn't `entityEvalValues` also contain errors?. Can we give this a more appropriate name?.

**Review by ohansFavour — COMMENTED** [review]
Can we add/update tests for addErrorToEntityProperty?

File: app/client/src/workers/Evaluation/dataTreeUtils.ts (hunk 2 above)
Comment (Rishabh-Rathod) [reply app/client/src/workers/Evaluation/dataTreeUtils.ts]: done.

File: app/client/src/workers/Evaluation/dataTreeUtils.ts (hunk 1 above)
Comment (Rishabh-Rathod) [reply app/client/src/workers/Evaluation/dataTreeUtils.ts:44]: I don't think it would be growing further.

**Review by ohansFavour — APPROVED** [review]

File: app/client/src/workers/Evaluation/dataTreeUtils.ts (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/workers/Evaluation/dataTreeUtils.ts:44]: All right

**Review by ohansFavour — APPROVED** [review]

**Review by AnandiKulkarni — APPROVED** [review]
